clone.go

```diff
@@ -0,0 +1,18 @@
+package fsutil
+
+import (
+	"fmt"
+
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
+)
+
+func cloneProto[T protoreflect.ProtoMessage](from T) (T, error) {
+	var zero T
+	clone := proto.Clone(from)
+	typed, ok := clone.(T)
+	if !ok {
+		return zero, fmt.Errorf("failed to cast %+v to %+v", clone, zero)
```

Comment:
only `pkg/errors` allowed in these repos

Reply:
I didn't know that. Will fix.